src/Kestrel.Core/Internal/Http/HttpHeaders.cs

```diff
@@ -170,6 +175,10 @@ protected virtual bool CopyToFast(KeyValuePair<string, StringValues>[] array, in
             {
                 ThrowHeadersReadOnlyException();
             }
+            if (string.IsNullOrEmpty(key))
+            {
+                throw new InvalidOperationException(CoreStrings.InvalidEmptyHeaderName);
```

Comment:
Use a not returning `ThrowXxx` method; rather than loading string and throwing inline.

Reply:
Also use same method as one above